libexec/mass-testing/test-harness

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python3
+
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import argparse
+import json
+import os
+import signal
+import sys
+from contextlib import suppress
+from ctypes import cdll
+from ctypes.util import find_library
+from pathlib import Path
+from shutil import which
+from subprocess import STDOUT, Popen, TimeoutExpired
+from tempfile import NamedTemporaryFile
+from time import sleep
+from typing import Collection, Generator, List
+
+import psutil
+
+SCRIPT_DIR = Path(__file__).parent.resolve()
+
+_time_json_format = (
+    '{"user_time": %U, "system_time": %S, "percent_cpu": "%P", "elapsed_time": %e, '
+    '"avg_shared_text_size": %X, "avg_unshared_data_size": %D, "avg_stack_size": %p, '
+    '"avg_total_size": %K, "max_resident_set_size": %M, "avg_resident_set_size": %t, '
+    '"major_io_page_faults": %F, "minor_reclaim_page_faults": %R, '
+    '"voluntary_context_switches": %w, "involuntary_context_switches": %c, "swaps": %W, '
+    '"fs_inputs": %I, "fs_outputs": %O, "socket_msgs_sent": %s, "socket_msgs_recvd": %r, '
+    '"signals_delivered": %k, "page_size": %Z, "exit_code": %x}'
+)
+
+
+def set_child_subreaper():
+    libc_name = find_library("c")
+    assert libc_name is not None
+    libc = cdll.LoadLibrary(libc_name)
+    libc.prctl(PR_SET_CHILD_SUBREAPER := 36, 1)  # noqa: F841,N806
+
+
+def kill(pid: int, signal: int):
+    with suppress(ProcessLookupError, PermissionError):
+        os.kill(pid, signal)
+
+
+def walk_process_tree(pid: int, include_root=True) -> Generator[psutil.Process, None, None]:
```

Comment:
Docs: does this yield root or leaves first?

Reply:
Added